### PR TITLE
Add four missing diagrams, provide contributor instructions, and update mermaid.

### PR DESCRIPTION
**Pull request by jtreher**
* The addon was missing support for MindMaps, Quadrants, XY, and Block-Beta. 
   * Added Support for those types.

* Added instructions for contributors to help out.
   * Let me know if you'd like me to change anything. I was a little iffy on the instructions. It's seems like there should be something more general purpose, but a lot of it is all about creating a plugin from scratch.

* Bumped mermaid to the latest version.

Screenshots of new diagrams rendering in my local vanilla Backstage instance:
image

image

image

image

Mermaid adds Katex support in [10.8](https://github.com/mermaid-js/mermaid/releases/tag/v10.9.0)

image

Fixes https://github.com/johanneswuerbach/backstage-plugin-techdocs-addon-mermaid/issues/54

### CODE REVIEW

**Comment by johanneswuerbach** [comment]
@jtreher this looks great, would you mind running `yarn install` locally and committing the resulting lockfile?

**Comment by jtreher** [comment]
@johanneswuerbach Ah, I was running yarn install, but it was only generating the lock file for the root Backstage folder. I had to remove the monorepo tooling.

Also, I bumped the package version.

It still seems like I need to do something.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "backstage-plugin-techdocs-addon-mermaid",
-  "version": "0.12.1",
+  "version": "0.13.0",
```
Comment (johanneswuerbach) [inline package.json:3]: Could you revert this change? The version is generally only increased once the PR has been merged into main.

Comment (jtreher) [reply package.json]: @johanneswuerbach Sorry about that. Thanks for your patience.

**Comment by johanneswuerbach** [comment]
Thank you for the contribution @jtreher 🤩 , published as `backstage-plugin-techdocs-addon-mermaid@0.13.0`.
